### PR TITLE
Computer vision example on MNIST

### PR DESCRIPTION
**Pull request by jbschiratti**
This PR aims at adding another computer vision example (training a FCN on MNIST data).
In #226, @sgugger agreed that such an example would be useful for Accelerate.
I would be happy to discuss this example (and finalize the PR if it still of interest).

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ To run it in each of these various modes, use the following commands:
 
 ## Simple vision example
 
-The [cv_example.py](./cv_example.py) script is a simple example to fine-tune a ResNet-50 on a classification task ([Ofxord-IIT Pet Dataset](https://www.robots.ox.ac.uk/~vgg/data/pets/)).
+The [cv_example.py](./pet_datase_example.py) script is a simple example to fine-tune a ResNet-50 on a classification task ([Ofxord-IIT Pet Dataset](https://www.robots.ox.ac.uk/~vgg/data/pets/)).
```
Comment (sgugger) [inline examples/README.md:102]: You should also mention you new example here :-)

File: examples/mnist_example.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+# in any of the following settings (with the same script):
+#   - single CPU or single GPU
+#   - multi GPUS (using PyTorch distributed mode)
+#   - (multi) TPUs
+#   - fp16 (mixed-precision) or fp32 (normal precision)
+#
+########################################################################
+
+
+from __future__ import print_function
```
Comment (sgugger) [inline examples/mnist_example.py:30]: We don't support Python 2, so there is no need for this.

File: examples/mnist_example.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                preds = output.argmax(dim=1, keepdim=True)
+
+                # Compute accuracy:
+                # We first gather predictions & labels from all processes
+                bs = _test_loader.batch_size
+                y_pred = accelerator.gather(preds).detach().cpu()[:bs]
+                y_true = accelerator.gather(target).detach().cpu()[:bs]
+                test_accuracy.update((y_pred, y_true))
+
+        test_acc = test_accuracy.compute()
```
Comment (sgugger) [inline examples/mnist_example.py:222]: I'd prefer to see all the logic to compute the accuracy here. We don't need a generic class for this in this particular example, and it would make it easier to read I believe.

File: examples/mnist_example.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+    # SAVE TRAINED MODEL (OPTIONAL)
+    if accelerator.is_main_process and args.out_dir is not None:
+        accelerator.wait_for_everyone()
+        unwrapped_model = accelerator.unwrap_model(model)
+        unwrapped_model.save(args.out_dir, save_function=accelerator.save)
+
+
+if __name__ == '__main__':
+
```
Comment (sgugger) [inline examples/mnist_example.py:258]:
```suggestion
```

**Review by sgugger — COMMENTED** [review]
Thanks for adding this example! I think it's great to showcase a pure PyTorch example with Accelerate.

I just have two remarks:
- it would be easier to read if the `evaluate` and `train_one_epoch` functions where outside of the main function, so they variables can be named without _. We can pass the `accelerator` object to those :-)
- it would also be easier to read if the code computing the accuracy was directly all inside the `evaluate` function, instead of relying on a class.
